### DISCUSSION TITLE
slowly componet compiling in JCB

### DISCUSSION THREAD

**Discussion by saman222** [joomengine/Joomla-Component-Builder#725] [Q&A]
hi
i create a component and when i want compile that, it's get more time to compile and some times get more than 200 second and return to me "ERR_EMPTY_RESPONSE".
i unlink all modules and plugins from component but it still get huge duration try to compile and in the end it can not finishing compoling process and return "ERR_EMPTY_RESPONSE".
i increase max_execution_time but have not any affect and manipulate other PHP configuration.
 can you advise me and how to resolve my problem?

**Comment by Llewellynvdm** [comment]
The first thing to do is to try compile the demo component, so we can see if that compiles. Normally this will show you that the problem is with your component setup, should the demo component compile without errors. Should it also have errors... then the problem is with your system setup.

Then depending on your components setup, it may need to access the internet to get or set things online, and if no connection is possible these kind of errors has been seen.

Next making sure all file and folders included in you component still exist, and are readable.

Also check that folder where JCB/Apache needs to write and create files and folders is set with the correct permissions. Which is the same [permissions](https://github.com/vdm-io/Joomla-Component-Builder/issues/682#issuecomment-794144104) as Joomla normally needs.
